test/runnable/inline7625.d

```diff
@@ -0,0 +1,150 @@
+// REQUIRED_ARGS: -inline
+
+/***************************************************/
+
+int foo1(int v)
+{
+    return bar1(2 * v);
+}
+
+int bar1(int a)
+{
+    if (a > 0)
+        return 1;
+    else
+        return baz1(a);
+}
+
+int baz1(int a)
+{
+    if (a > 0)
+        throw new Exception("a > 0");
+    return a - 1;
+}
+
+// ---
+
+int foo2(int v)
+{
+    return bar2(2 * v);
+}
+
+int bar2(int a)
+{
+    if (a > 0)
+        return 1;
+    // else
+        return baz2(a);
+}
+
+int baz2(int a)
+{
+    if (a > 0)
+        throw new Exception("a > 0");
+    return a - 1;
+}
+
+void test7625a()
+{
+    assert(foo1(1) ==  1);
+    assert(foo1(0) == -1);
+    assert(foo2(1) ==  1);
+    assert(foo2(0) == -1);
+}
+
+/***************************************************/
+
+int tembo(int x, int y)
+{
+    if (y == 0)
+        return 0;
+    x++;
+    return x / y;
+}
+int pembo(int x, int y)
+{
+    if (y == 0)
+        return 0;
+    else
+    {
+        x++;
+        return x / y;
+    }
+}
+
+int twiga(int x, int y, int z)
+{
+    auto w = tembo(x, y);
+    return w * z;
+}
+
+int simba(int x, int y, int z)
+{
+    auto w = pembo(x, y);
+    return w * z;
+}
+
+void test7625b()
+{
+    assert(twiga(5, 3, 4) == 8);
+    assert(twiga(5, 0, 4) == 0);
+
+    assert(simba(5, 3, 4) == 8);
+    assert(simba(5, 0, 4) == 0);
+}
+
+/***************************************************/
+
+@safe pragma(inline, true)
+{
+    bool inlineMe15483a(bool left)
+    {
+        if (left)
+            return true;
+
+        return false;
+    }
+
+    bool inlineMe15483b(bool left)
+    {
+        if (left)
+            return true;
+
+        static if (false)
+        {
+            /*
+                Even though it does absolutely nothing,
+                the mere presence of this block prevents inlining
+                of this function.
+            */
```

Comment:
Hmm.  Is that by design?

Reply:
This is the bug in question being tested.

Reply:
Or at least one aspect of it if I understand right.

Reply:
IMO this needs a better comment.  It should read something to the effect of "Ensure that the presence of a `static if` block does not prevent inlining.  See {insert bug URL}."  I understand that the original comment was to clarify the problem, but now we need to clarify what we're testing.